src/Storages/MergeTree/MergeTreeIndexFullText.h

```diff
@@ -190,6 +190,20 @@ struct SplitTokenExtractor final : public ITokenExtractor
     bool supportLike() const override { return true; }
 };
 
+/// Parser extracting all ngrams token from string.
+struct NgramTkTokenExtractor final : public ITokenExtractor
+{
+    NgramTkTokenExtractor(size_t n_) : n(n_) {}
+
+    static String getName() { return "ngramtkbf_v1"; }
```

Comment:
Why did you choose that name? Is it something standard?
Otherwise we better make it more verbose.

Reply:
Sorry, I just want to combine ngram and token index, 'tk' standard for 'token'. I come from China, we try use ClickHouse to store app logs. We have a lot of logs contruct with chinese words and english words. I try to indexing chinese words with ngram and indexing english words with token. But this pull request has a bug, which is different with tokenbf_v1 handling '%something%'.

Reply:
We are not trying to make function names as short as possible. You can try not to shorten Token, Ngram etc and make it more descriptive. And version in name looks also strange. Better add a word or more that shows what is specific here

Reply:
OK, I will close this pull request first.